File: csm/report.py
"""Appends the per-item run report: ~/.csm/report.md."""

from datetime import datetime

from . import config


def append_run_header(trigger: str) -> None:
    _append(f"\n## Run {datetime.now():%Y-%m-%d %H:%M} ({trigger})\n")


def append_item(item: dict, status: str, session_id: str | None,
                summary: str | None, error: str | None, weighted_tokens: int,
                weekly_spend: int, budget: int) -> None:
    lines = [
        f"### [{status}] {item['id']} - {_short(item['prompt'])}",
        f"- project: `{item['project']}`",
    ]
    if session_id:
        lines.append(f"- session: `{session_id}` (resume with `claude -r {session_id}`)")
    if summary:
        lines.append(f"- summary: {summary}")
    if error:
        lines.append(f"- error: {error}")
    lines.append(f"- tokens (weighted): {weighted_tokens:,} | week: {weekly_spend:,} / {budget:,}")
    _append("\n".join(lines) + "\n")


def append_note(text: str) -> None:
    _append(f"- {text}\n")


def _append(text: str) -> None:
    path = config.report_path()
    path.parent.mkdir(parents=True, exist_ok=True)
    with path.open("a", encoding="utf-8") as fh:
        fh.write(text)


def _short(prompt: str, limit: int = 80) -> str:
    line = prompt.strip().splitlines()[0] if prompt.strip() else ""
    return line[:limit] + ("..." if len(line) > limit else "")

File: internal/report/report.go
// Package report appends the per-item run report to ~/.csm/report.md.
package report

import (
	"fmt"
	"os"
	"path/filepath"
	"strings"
	"time"

	"github.com/EkinBarisC/claude-session-manager/internal/config"
	"github.com/EkinBarisC/claude-session-manager/internal/queue"
)

func AppendRunHeader(trigger string) {
	appendText(fmt.Sprintf("\n## Run %s (%s)\n", time.Now().Format("2006-01-02 15:04"), trigger))
}

func AppendItem(item *queue.Item, status, sessionID, summary, errMsg string, weightedTokens, weeklySpend, budget int, logPath string) {
	lines := []string{
		fmt.Sprintf("### [%s] %s - %s", status, item.ID, Short(item.Prompt, 80)),
		fmt.Sprintf("- project: `%s`", item.Project),
	}
	if item.Branch != "" {
		lines = append(lines, fmt.Sprintf("- branch: `%s`", item.Branch))
	}
	if sessionID != "" {
		lines = append(lines, fmt.Sprintf("- session: `%s` (resume with `claude -r %s`)", sessionID, sessionID))
	}
	if summary != "" {
		lines = append(lines, "- summary: "+summary)
	}
	if logPath != "" {
		lines = append(lines, fmt.Sprintf("- transcript: `%s`", logPath))
	}
	if errMsg != "" {
		lines = append(lines, "- error: "+errMsg)
	}
	lines = append(lines, fmt.Sprintf("- tokens (weighted): %d | week: %d / %d", weightedTokens, weeklySpend, budget))
	appendText(strings.Join(lines, "\n") + "\n")
}

func AppendNote(text string) {
	appendText("- " + text + "\n")
}

func appendText(text string) {
	path := config.ReportPath()
	os.MkdirAll(filepath.Dir(path), 0o755)
	f, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
	if err != nil {
		return
	}
	defer f.Close()
	f.WriteString(text)
}

// Short returns the first line of text, truncated to limit runes.
func Short(text string, limit int) string {
	line := ""
	for _, l := range strings.Split(strings.TrimSpace(text), "\n") {
		line = strings.TrimSpace(l)
		break
	}
	runes := []rune(line)
	if len(runes) > limit {
		return string(runes[:limit]) + "..."
	}
	return line
}
